src/commonTest/kotlin/com/github/quillraven/fleks/Fleks2TDD.kt

```diff
@@ -20,6 +20,29 @@ private data class Position(
     companion object : ComponentType<Position>()
 }
 
+abstract class ColliderService {
+    abstract fun getId(): String
+}
+
+private data class Collider(
+    val size: Float
+) : Component<Collider> {
+    override fun type(): ComponentType<Collider> = Collider
+
+    var colliderId: String? = null
+
+    override fun World.onAddComponent() {
+        val provider = inject<ColliderService>()
+        colliderId = provider.getId()
+    }
+
+    override fun World.onRemoveComponent() {
+        colliderId = null
+    }
+
+    companion object : ComponentType<Collider>()
+}
```

Comment:
Since component hooks could be defined per-test and component lifecycle methods can't, I figured a separate component would be appropriate to test the specific behaviour instead of retrofitting existing ones. I also included an injectable to make it more of an E2E use-case for the lifecycle methods.

Reply:
Good idea - thanks!